src/main/java/pl/palonek/interview/course/service/CourseSummaryService.java

```diff
@@ -0,0 +1,64 @@
+package pl.palonek.interview.course.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.palonek.interview.course.model.CourseSession;
+import pl.palonek.interview.course.model.CourseSummary;
+import pl.palonek.interview.course.repository.CourseRepository;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CourseSummaryService {
+
+    private final CourseRepository courseRepository;
+
+    public CourseSummary getCourseSummary(String userId, String courseId) {
+        List<CourseSession> sessions = courseRepository.findByUserIdAndCourseId(userId, courseId);
+
+        int maxScore = getMaxScore(sessions);
+        double average = getAverage(sessions);
+        ZonedDateTime startDate = getSessionsStartDate(sessions);
+        ZonedDateTime endDate = getSessionsEndDate(sessions);
+        long totalSessionTime = getTotalTimeBetweenSessions(startDate, endDate);
```

Comment:
[subjective] Maybe it would violate YAGNI, but I would move get* methods from this services to specialized class which would operate on List<CourseSession>. 

Like this:
foo = new CourseSessionStats(session);
foo.getAverage();
foo.getCourseSummary().